### ISSUE TITLE
Some pngs don't display under NixOS

### ISSUE THREAD

**Issue by RamKromberg** [lenmus/lenmus#50]
I've recently [packaged lenmus in nixos](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/lenmus/default.nix) and noticed that some pngs fail to open and even throw exceptions.
For instance, when opening Music Reading Level 1, _cc-by-sa.png_ fails to open and the following error shows:
```Line 0. Error loading image. Non-standard unknown exception```
in the shell, the following also appears:
```
(lenmus:23951): Gtk-WARNING **: Unable to show 'Open the music books': Operation not supported
Non-standard unknown exception (catch in ImageReader::load_image)
```

The strange thing is that it only happens with some images: In the same book, under _The Tempo and Metronome_ page, the _metronome_image.png_ image loads fine.

Anyhow, it's probably related to the package or some dependency. But I figured I'd report it just-in-case this is a known issue or something similar.

**Comment by cecilios** [comment]
Thanks for reporting this. The issue with png images is strange and I cannot reproduce this. A few months ago there were an issue with the eBooks after fixing some translations. But this was solved. A couple of questions:

- The cc-by-sa.png that fails is common to all eBooks. Does the crash happen only with Music Reading Level 1 eBook? Can you check with the other eBooks please?

- Which LenMus and Lomse versions are you using? You can check both in Main menu > Help > About > Build info. button.

Thanks

**Comment by RamKromberg** [comment]
> Does the crash happen only with Music Reading Level 1 eBook?

0. It doesn't crash. The exception is caught and the lenmus keeps running and working fine but with a gray box instead of a png. That is, it's just an error message box and an insignificant graphical glitch. Otherwise, I wouldn't have packaged it ;)

1. The exception is thrown with all the ebooks like you say. However, it only fails on the specific pages that contain that specific png. That is, every first page and the "Credits and Copyrights" pages.

> Which LenMus and Lomse versions are you using?

```
Build information
Program build date: Jan 1 1970
wxWidgets 3.0.2
lomse 0.16.2 (rev.127)
wxMidi 2.0
sqlite3 3.13.0
Your computer information:
Charset encoding: UTF-8
```

You can see in the package deceleration (https://github.com/NixOS/nixpkgs-channels/blob/nixos-unstable/pkgs/applications/misc/lenmus/default.nix) that it fetches the Release_5.4.1 tag off github (https://github.com/lenmus/lenmus/archive/Release_5.4.1.tar.gz). 

Also, the png++ package version is 0.2.9 (https://github.com/NixOS/nixpkgs-channels/blob/nixos-unstable/pkgs/development/libraries/png++/default.nix) and the libpng version is 1.6.23 (https://github.com/NixOS/nixpkgs-channels/blob/nixos-unstable/pkgs/development/libraries/libpng/default.nix).

**Comment by cecilios** [comment]
Thanks for the info.

In any case, the LenMus code that you are using is old, but probably this is may fault as I do not create releases, but just update the main repository, expecting that users will build from latest code. I have to change this and create releases more often.

I will investigate more the reported issues and, in any case, I will create a new release in a couple of days.

**Comment by RamKromberg** [comment]
I'll try building the git master and let you know if the issue persists.

**Comment by RamKromberg** [comment]
@cecilios  Ok I found out what's wrong in the .png and fixed it in the commit https://github.com/lenmus/lenmus/issues/51.

I believe the problem was the embedded colorspace. When I compared cc-by-sa.png to the rest of the pngs, I noticed it was the only one with one. Anyhow, I fixed the problem by resizing another png and copy-pasting the pixels into that.

I didn't regenerate the books in books/src/. Instead, I tested with the following:
1. Copied the .lmb.
2. Changed the extension to .zip.
3. Replaced cc-by-sa.png with the one in https://github.com/lenmus/lenmus/commit/29c1d8cb8266e8d9f140f64570af343089f91b1a.
4. Renamed extension to .lmb.
5. Opened the book with lenmus and verified there wasn't an exception.

I've also addressed two other issues that I'll explain in the PR.
